### PR TITLE
[FEATURE] Change Default CLI Flag To V3

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* While the flag `--v3-api` is still accepted, it is now optional; in other words, omitting the API version flag defaults to `V3` CLI API.  It is now required to specify `--v2-api` to invoke the legacy `V2` CLI API behavior.
* Tests have been updated as follows: the majority of CLI tests still specify `--v3-api` option; however, this option has been removed from the CLI usage in integration tests.  This verifies both, the explicit and the default invocation of CLI `V3` API.

### Next Work
* Documentation (including DocuSaurus-enabled documentation under test), in accordance with the changes in #3938 and in the present work, still has to be developed in order for the user guides to stay consistent with the system behavior.
* UAT (User Acceptance Testing) is highly recommended for the encompassing change, afforded by the present work.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- JIRA: GREAT-286
-
-

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/cli/cli.py [hunk 1]
```diff
@@ -88,27 +88,27 @@ def print_ctx_debugging(ctx):
         print(f"ctx.find_root().protected_args: {ctx.find_root().protected_args}")
 
     @staticmethod
-    def is_v2_api(ctx):
-        """Determine if v2 api is requested by searching context params."""
+    def is_v3_api(ctx):
+        """Determine if v3 api is requested by searching context params."""
         if ctx.params:
-            return ctx.params and "v2_api" in ctx.params.keys() and ctx.params["v2_api"]
+            return ctx.params and "v3_api" in ctx.params.keys() and ctx.params["v3_api"]
```
Comment (cdkini) [inline great_expectations/cli/cli.py:94]: Should this possibly be `ctx.params["v3_api"]`? Or are we always guaranteed it and just checking for nulls?

Would there be any benefit in using `get()`?

**Review by Rachel-Reverie — APPROVED** [review]

**Comment by Shinnnyshinshin** [comment]
Hi @alexsherstinsky thank you very much for this 🙇🏼  This may be betraying my ignorance on our CLI-Code, but are there any CLI-tests that need to be added or updated to take in the `--V2-API` flag? 

Like `great_expectations --V2-API do_a_v2_thing` --> `[ yes i did a V2 thing and not a V3 thing]`

File: great_expectations/cli/cli.py [hunk 2]
```diff
@@ -88,27 +88,27 @@ def print_ctx_debugging(ctx):
...
 @click.version_option(version=ge_version)
 @click.option(
-    "--v2-api/--v3-api",
-    "v2_api",
+    "--v3-api/--v2-api",
+    "v3_api",
     is_flag=True,
     default=True,
-    help="Default to v2 (Batch Kwargs) API. Use --v3-api for v3 (Batch Request) API",
+    help="Default to v3 (Batch Request) API. Use --v2-api for v2 (Batch Kwargs) API",
```
Comment (NathanFarmer) [inline great_expectations/cli/cli.py:111]: ❤️

**Review by NathanFarmer — APPROVED** [review]
LGTM

**Comment by alexsherstinsky** [comment]
> Hi @alexsherstinsky thank you very much for this 🙇🏼 This may be betraying my ignorance on our CLI-Code, but are there any CLI-tests that need to be added or updated to take in the `--V2-API` flag?
> 
> Like `great_expectations --V2-API do_a_v2_thing` --> `[ yes i did a V2 thing and not a V3 thing]`

@Shinnnyshinshin This is actually a very good question, and I spent ample time last night on ascertaining the associated behavior.  In a nutshell, our `V2` API just works in the `v012` subdirectory.  The main `great_expectation` CLI call comes into the `cli.py` code in the directory above, which routes the execution to the respective implementation.  Hence, no additional flags must be supplied to the `V2` tests.  Thanks!

**Review by cdkini — APPROVED** [review]

File: great_expectations/cli/cli.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/cli/cli.py:94]: @cdkini I did not want to change this aspect, since it has worked for over a year reliably.

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM :) Thank you for the additional clarity around my question

**Review by anthonyburdi — APPROVED** [review]
LGTM

**Pull request merged**